tests/integration/helpers.py

```diff
@@ -151,12 +151,44 @@ def project_exists(self, project_name):
         if location is _sentinel:
             location = self.example_repo
         repository_id = self.uuid4()
-        repository_iri = self.resource("repositories", repository_id)
+        repository_iri = self.resource("stacks", repository_id)
         await self.insert_node(Node(repository_iri, {
-            RDF.type: Doap.GitRepository,
+            RDF.type: Doap.Stack,
             Mu.uuid: repository_id,
             Doap.location: location,
         }))
+
+        drc_id = self.uuid4()
+        drc_text = """
+            version: \"2\"
+            services:
+
+              service1:
+                image: busybox
+                command: \"sleep 60\"
+
+              service2:
+                image: busybox
+                command: \"sleep 60\"
+        """
+        d_iri = IRI("http://stack-builder.big-data-europe.eu/resources/"
+        drc_iri = d_iri + "%s/%s" % ("docker-composes", drc_id)
+        drc_title = "stack_{}_drc_{}".format(repository_id, drc_id)
```

Comment:
Avoid mixing string formatting mechanisms, especially inside the same function. Some people tends to avoid it for an entire project but I believe every formatting mechanism has its own benefits.

```python
# using %
assert ("foo%sbaz" % "bar") == "foobarbaz"
# using .format
assert "foo{}baz".format("bar") == "foobarbaz"
# or the all new f-string of Python 3.5
x = "bar"
assert f"foo{x}baz" == "foobarbaz"
```

None of them are going to be deprecated.

https://stackoverflow.com/questions/43123408/f-strings-in-python-3-6